dodiscover/context.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional, Set, Union
+
+import networkx as nx
+import pandas as pd
+
+from ._protocol import GraphProtocol
+
+
+class Context:
+    """Context of assumptions, domain knowledge and data.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        A dataset, consisting of samples as rows and columns as variables.
+    variables : Optional[Set], optional
+        Set of observed variables, by default None. If neither ``latents``,
+        nor ``variables`` is set, then it is presumed that ``variables`` consists
+        of the columns of ``data`` and ``latents`` is the empty set.
+    latents : Optional[Set], optional
+        Set of latent "unobserved" variables, by default None. If neither ``latents``,
+        nor ``variables`` is set, then it is presumed that ``variables`` consists
+        of the columns of ``data`` and ``latents`` is the empty set.
+    init_graph : Optional[GraphProtocol], optional
+        The graph to start with, by default None.
+    included_edges : Optional[nx.Graph], optional
+        Included edges without direction, by default None.
+    excluded_edges : Optional[nx.Graph], optional
+        Excluded edges without direction, by default None.
+
+    Raises
+    ------
+    ValueError
+        ``variables`` and ``latents`` if both set, should contain the set of
+        all columns in ``data``.
+
+    Notes
+    -----
+    Context is a data structure for storing assumptions, domain knowledge,
+    priors and other structured contexts alongside the datasets. This class
+    is used in conjunction with a discovery algorithm.
+
+    Setting the apriori explicit direction of an edge is not supported yet.
+    """
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        variables: Optional[Set] = None,
+        latents: Optional[Set] = None,
```

Comment:
How would we use the ```latents``` information here? Would it be sufficient to simply indicate between which variables we expect latent confounders?

Reply:
Just following the API here: https://github.com/py-why/dowhy/wiki/API-for-Global-Causal-Discovery

I think latents would not normally be used unless maybe doing some type of sensitivity analysis where you assume all of a sudden "x" is unobserved